packages/runtime-common/loader.ts

```diff
@@ -435,11 +435,20 @@ export class Loader {
     urlOrRequest: string | URL | Request,
     init?: RequestInit,
   ): Promise<Response> {
+    let mergedHeaders: HeadersInit =
+      urlOrRequest instanceof Request
+        ? headersFromRequest(urlOrRequest)
+        : init?.headers ?? {};
     try {
       for (let handler of this.urlHandlers) {
         let request = this.asRequest(urlOrRequest, init);
+        mergedHeaders = { ...mergedHeaders, ...headersFromRequest(request) };
+        setHeaders(request, mergedHeaders);
 
         let result = await handler(request);
+        // the handler is allowed to mutate the headers, so we merge any updated headers
+        mergedHeaders = { ...mergedHeaders, ...headersFromRequest(request) };
```

Comment:
If its just realm auth headers, then can we make it specific as what header is being set by the handler?

Reply:
Look, it’s not it’s a completely different header that has nothing to do with any th and everything to do with indexing. Read comment above.

Reply:
I think it’s a bad idea to put header specific logic at this level. Header spec logic needs to live in the url handler itself—otherwise the abstraction starts leaking out